### PR TITLE
Locales for quests

### LINKED ISSUE #207: locales_ translations in database dont work properly with core? esES Client
**Description**: Locales translations don't work properly (for me)

**Current behaviour**: I have a database of the same exact estructure than the one using with azerothcore-wotlk, i translated everything as it is in the other database i use with an old Trinitycore compilation,, but in this case with azerothcore-wotlk.. quests dont detect the translations on locales_ tables on world database,, items dont translate, npc texts dont translate,, they still remain in english, i used locales 6 as in my old database that worked properly, this is maybe a think i should set-up in worldserver.conf file? I did something wrong?
Its problem of the core not reconizing the locales?
Should i help with a fix to this?

**Expected behaviour**: Using translation locales_ system should translate for clients texts to language according to his client, in my case esES.

**Operating system**:  Windows 10 Pro

**Modules**:  Should i use any module to solve this? Don't really know...

**Other Customizations**: Clean compile, everything works fine except for translations to Spanish.

If you can explain me detaile

If you need screenshots i can attach them.
Thanks and sorry for my english.
Keep this great work.
##

### PR DESCRIPTION
**Pull request by Viste**
**Changes proposed:**

-  Restore Locale_quest system load from db

**Target branch(es):** 1.x/2.x etc.
master
**Issues addressed:** Closes #207  (only for quests, npc_text need to implent to)

**Tests performed:** (Does it build, tested in-game, etc)
build and test in game all ok for me
**Known issues and TODO list:**
no one
thx to trinity core commit history :D

### CODE REVIEW

**Comment by Viste** [comment]
All done need only table have only few insertion  to test screens: 
wowscrnshot_030217_183831
wowscrnshot_030217_183841

**Comment by fr4z3n** [comment]
Thank you for converting it to azerothcore.

Can we merge this?
 👍

File: src/game/Entities/Creature/GossipDef.cpp [hunk 1]
```diff
@@ -280,13 +294,31 @@ void PlayerMenu::SendQuestGiverStatus(uint8 questStatus, uint64 npcGUID) const
...
+	std::string questDetails = quest->GetDetails();
+	std::string questObjectives = quest->GetObjectives();
+	std::string questEndText = quest->GetEndText();
+
+	int32 locale = _session->GetSessionDbLocaleIndex();
+	if (locale >= 0)
+	{
+		if (QuestLocale const* localeData = sObjectMgr->GetQuestLocale(quest->GetQuestId()))
+		{
+			ObjectMgr::GetLocaleString(localeData->Title, locale, questTitle);
```
Comment (fr4z3n) [inline src/game/Entities/Creature/GossipDef.cpp]: Tabs?

File: src/game/Entities/Creature/GossipDef.cpp [hunk 2]
```diff
@@ -368,17 +400,158 @@ void PlayerMenu::SendQuestGiverQuestDetails(Quest const* quest, uint64 npcGUID,
...
 void PlayerMenu::SendQuestGiverOfferReward(Quest const* quest, uint64 npcGUID, bool enableNext) const
 {
+	//load from db why sunwell remove it ??
+	std::string questTitle = quest->GetTitle();
+	std::string questOfferRewardText = quest->GetOfferRewardText();
+
+	int32 locale = _session->GetSessionDbLocaleIndex();
+	if (locale >= 0)
+	{
+		if (QuestLocale const* localeData = sObjectMgr->GetQuestLocale(quest->GetQuestId()))
```
Comment (fr4z3n) [inline src/game/Entities/Creature/GossipDef.cpp]: tabs

File: src/game/Entities/Creature/GossipDef.cpp [hunk 3]
```diff
@@ -454,6 +627,20 @@ void PlayerMenu::SendQuestGiverRequestItems(Quest const* quest, uint64 npcGUID,
...
     // items.  Otherwise, we'll skip straight to the OfferReward
+	
+	//again db load
+	std::string questTitle = quest->GetTitle();
+	std::string requestItemsText = quest->GetRequestItemsText();
+
+	int32 locale = _session->GetSessionDbLocaleIndex();
+	if (locale >= 0)
+	{
+		if (QuestLocale const* localeData = sObjectMgr->GetQuestLocale(quest->GetQuestId()))
```
Comment (fr4z3n) [inline src/game/Entities/Creature/GossipDef.cpp]: tabs

**Comment by cooler-SAI** [comment]
Very good Job!! I hope this can be merged Quick!))))

**Comment by ShinDarth** [comment]
@cooler-SAI  you can test this and give your feedback in order to speed up the process :D

**Comment by HeartWell** [comment]
It looks right and works.

**Comment by ShinDarth** [comment]
I guess it can be merged as soon as the comments from @fr4z3n are resolved

**Comment by fr4z3n** [comment]
Should be resolved now @ShinDarth

**Pull request merged**